src/usage_metrics/core/eel_hole.py

```diff
@@ -0,0 +1,366 @@
+    timestamp: datetime.datetime
+    text_payload: str | None = None
+
+    @field_validator("*", mode="before")
+    def replace_na_with_none(cls, value):  # noqa: N805
+        """To successfully validate string dtypes, convert NaNs to None."""
+        if isinstance(value, float) and math.isnan(value):
+            return None
+        return value
+
+    @model_validator(mode="before")
+    def drop_non_events(cls, data):  # noqa: N805
+        """Where JSON payload event is a 'loading' message, drop JSON payload."""
+        if isinstance(data["jsonPayload"], dict):  # noqa: SIM102
+            if (
+                event := data["jsonPayload"].get("event")
+            ) and "loading datapackage from" in event:
+                data.pop("jsonPayload", None)
+        return data
+
+    class Config:  # noqa: D106
+        alias_generator = to_camel
+        populate_by_name = True
+
+
+@asset(
+    partitions_def=WeeklyPartitionsDefinition(start_date="2023-08-16"),
+    tags={"source": "eel_hole"},
+)
+def _core_eel_hole_logs(
+    context: AssetExecutionContext,
+    raw_eel_hole_logs: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform viewer.catalyst.coop logs."""
+    context.log.info(f"Processing data for the week of {context.partition_key}")
+
+    if raw_eel_hole_logs.empty:
+        context.log.warn(f"No data found for the week of {context.partition_key}")
+        return raw_eel_hole_logs
+
+    # Flatten the many nested columns and coerce them into the expected class
+    # Then drop the two columns (json_payload and json_payload_params) that we exploded
+    # into many other columns and are now empty
+    models = [
+        EelHoleLogs(**row).model_dump()
+        for row in raw_eel_hole_logs.to_dict(orient="records")
+    ]
+    converted_df = pd.json_normalize(models, sep="_").drop(
+        columns=["json_payload", "json_payload_params"]
+    )
+
+    # Also drop some columns that just provide constant metadata about the GCS logging
+    # instance
+    converted_df = converted_df.drop(
+        columns=[
+            "log_name",
+            "labels_instanceId",
+            "resource_type",
+            "resource_labels_configuration_name",
+            "resource_labels_location",
+            "resource_labels_project_id",
+            "resource_labels_revision_name",
+            "resource_labels_service_name",
+            "resource_labels_service_name",
+        ]
+    )
+
+    # JSON payload timestamp is least complete, and receive timestamp just
+    # tells us when the cloud run server received the sent logs (not that interesting).
+    # These vary by sub-seconds, so we'll just pick the standard 'timestamp'.
+    # See https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#FIELDS.timestamp
+    converted_df = converted_df.drop(
+        columns=["json_payload_timestamp", "receive_timestamp"]
+    )
+
+    # The filters are a list of dictionaries, so we manually split these out into
```

Comment:
non-blocking. i could not be seeing something in here that would not make this work but it seems like this could be migrated into the filters table.